### DISCUSSION TITLE
Removing upper() function in providers

### DISCUSSION THREAD

**Discussion by konart** [famiu/feline.nvim#191] [Ideas]
This is a breaking change suggestion but still:

I think it would be much better if default providers wouldn't use `upper()` function (see https://github.com/feline-nvim/feline.nvim/blob/4c6b9887f7e7ca977dd64ed57861102c0e778b8b/lua/feline/providers/file.lua#L147 for example).

This assumes that everyone use upper case in their status which is obviously not the case.
Converting to the lower case is not a problem, but this kill the original style in some cases (a buffer called `NvimTree` for example)

**Comment by lukas-reineke** [comment]
The providers are opinionated. We can't make everyone happy. If we make them lowercase, someone else will complain.

You can copy the providers and adjust to how you want it to be displayed.

**Reply by konart** [reply]
That's the thing - if feline default provider returns the string unchanged - client can adjust them as they wish or leave it be.
However you can't go black to NvimTree from the upper cased version.

**Reply by lukas-reineke** [reply]
If we return them unchanged, we force all users to have an extra function to adjust them, or they have inconsistent cases.

Sorry, but if you want them unchanged, just copy the providers into your config and make the change there. I think the default should stay like it is.
